app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    [
+        Person(person_dict["name"], person_dict["age"])
+        for person_dict in people_list
+    ]
+
+    for person_dict in people_list:
+        person = Person.people[person_dict["name"]]
+        if spouse_name := person_dict.get("wife"):
+            person.wife = spouse = Person.people[spouse_name]
+            spouse.husband = person
```

Comment:
There is no need to perform this operation since `husband `would be handled separately in other if condition. Remove it here and in the other if condition as well